cmake/AutoRevision.cmake

```diff
@@ -0,0 +1,73 @@
+# This file runs a collection of git commands to help automate the process
+# of generating the build revision. It takes an update to 3 files by hand
+# to do it manually and automates it.
+
+# Auto Revision needs git to work
+find_package(Git)
+
+if(Git_FOUND)
+
+  # Run git status to see if we are in a real local repo
```

Comment:
Can you specify `--git-dir` in case someone extracts a tarball within some other git-manage area?

Reply:
Pushed another update